### PR TITLE
Replace cxxopts with CLI11

### PR DESCRIPTION
**Pull request by mcfadden8**
New fan of CLI11

### CODE REVIEW

**Comment by rhornung67** [comment]
@mcfadden8 your last commit is very stylish...

**Comment by mcfadden8** [comment]
LGTM

**Comment by mcfadden8** [comment]
LGTM

File: tests/tpl/cli11_smoke_test.cpp [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+{
+  std::stringstream sstr;
+
+  sstr << CLI11_VERSION;
+
+  return sstr.str();
+}
+
+//-----------------------------------------------------------------------------
+int main(int argc, char* argv[])
```
Comment (davidbeckingsale) [inline tests/tpl/cli11_smoke_test.cpp:24]: Might be good to have some gtest assertions in here for stuff that should definitely be true.

Comment (mcfadden8) [reply tests/tpl/cli11_smoke_test.cpp:24]: Agreed.  I need to find some examples someplace of a gtest way of passing in command line arguments to be parsed.  This is presently just being called directly.

**Comment by mcfadden8** [comment]
Ruh-roh...  All things look good except for one of the builds (Optimized XL).  io_test is failing.  Need to investigate that more.  It is strange because it is working everywhere else.

**Comment by mcfadden8** [comment]
LGTM

File: tests/tpl/cli11_smoke_test.cpp (hunk 1 above)
Comment (mcfadden8) [reply tests/tpl/cli11_smoke_test.cpp]: Done

**Comment by mcfadden8** [comment]
LGTM

**Pull request merged**